.github/workflows/black.yml

```diff
@@ -0,0 +1,17 @@
+name: "Black Code Formatter"
+
+on:
+  push:
+    branches:
+      - 'release-*'
+  pull_request:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: psf/black@stable
```

Comment:
Come to think of it, ASF/Apache seems to have some rules about external **action** references to use, and those that don't meet the requirements be disabled (I found ↓)

<img width="1135" alt="image" src="https://github.com/apache/incubator-hugegraph-ai/assets/17706099/74cb7163-0e57-406f-8d29-6376529a7b36">

https://infra.apache.org/github-actions-policy.html

Reply:
Right, I'll check that later~

Reply:
![image](https://github.com/apache/incubator-hugegraph-ai/assets/58508660/66b52103-b7ee-47fd-8472-0c6d1f17147c)
that is why Black action is not showing

Reply:
> ![image](https://private-user-images.githubusercontent.com/58508660/336337512-66b52103-b7ee-47fd-8472-0c6d1f17147c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0OTA4NzAsIm5iZiI6MTcxNzQ5MDU3MCwicGF0aCI6Ii81ODUwODY2MC8zMzYzMzc1MTItNjZiNTIxMDMtYjdlZS00N2ZkLTg0NzItMGM2ZDFmMTcxNDdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA0VDA4NDI1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3MWJjODI5ZTcwNGY0YTE0NGJlODU5Njg2YmUzNTUzNWZjYmE5NjQwNzZmOTI5NWU3ZWUwNTdjZGZkMzlkMjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LngfZCygXuGSUosuGtKiyFvPe7fa_Q4igkVulX1ZVEM) that is why Black action is not showing

@liuxiaocs7 @simon824 any suggestion?

Reply:
I think we have 2 approaches:
1. follow the rules and use the specific git hash (SHA1) of the action, for example `psf/black@3702ba224ecffbcec30af640c149f231d90aebdb`
2. use `pip install & bash` as https://github.com/apache/incubator-hugegraph-ai/blob/main/.github/workflows/pylint.yml did

I prefer approach 1.

Also, i found a  ASF repo which may not obey the restriction but still confused how they did it.
In https://github.com/apache/shenyu/blob/master/.github/workflows/e2e-k8s.yml
![image](https://github.com/apache/incubator-hugegraph-ai/assets/58508660/9bdad9fc-32f2-4f57-9735-62bf1ab7e819)

Reply:
> I think we have 2 approaches:
> 
> 1. follow the rules and use the specific git hash (SHA1) of the action, for example `psf/black@3702ba224ecffbcec30af640c149f231d90aebdb`
> 2. use `pip install & bash` as [`main`/.github/workflows/pylint.yml](https://github.com/apache/incubator-hugegraph-ai/blob/main/.github/workflows/pylint.yml?rgh-link-date=2024-06-05T05%3A26%3A15Z) did
> 
> I prefer approach 1.
> 
> Also, i found a ASF repo which may not obey the restriction but still confused how they did it. In [apache/shenyu@`master`/.github/workflows/e2e-k8s.yml](https://github.com/apache/shenyu/blob/master/.github/workflows/e2e-k8s.yml?rgh-link-date=2024-06-05T05%3A26%3A15Z) ![image](https://private-user-images.githubusercontent.com/58508660/336698109-9bdad9fc-32f2-4f57-9735-62bf1ab7e819.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1NzY4MDYsIm5iZiI6MTcxNzU3NjUwNiwicGF0aCI6Ii81ODUwODY2MC8zMzY2OTgxMDktOWJkYWQ5ZmMtMzJmMi00ZjU3LTk3MzUtNjJiZjFhYjdlODE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDA4MzUwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjYjE4NmI5OGU1MGJhOTRkMWZhNTczNmI2MjUwYTU4NTMzOTMzYTg4NTFhNWQ1ZjM2ZmI1OWZkZDMxMDRjYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Q1nvhI9eVfoS_sgrxtLJ8Gz8Pyo8L_gcS0E_BNeUg6A)

Thanks for ur feedback~

U could try way1 first (This seems to be a new rule added this year, at least I didn't see it last year)